### PR TITLE
Reorg lib structure, added adafruit sensor func

### PR DESCRIPTION
**Pull request by ShyamHari**
Can now read w/ new IMU + reorg lib structure

### CODE REVIEW

File: Arduino/libraries/sensors/sensors.cpp [hunk 1]
```diff
@@ -3,15 +3,30 @@
...
+long dt; // change in time actual (milliseconds)
+const float minGyroValue = 0.25; // min +/-Gyro value, (converted to rad/s)
 // ============== Public Methods ==============
 
 Sensors::Sensors(int trig_pin, int echo_pin[3], int num_sensors){
   _trig_pin = trig_pin;
   _echo_pin = echo_pin;
   _num_sensors = num_sensors;
+  measurement_delay_us = 65535;
+  previousMillis = 0;
```
Comment (ParthSareen) [inline Arduino/libraries/sensors/sensors.cpp]: nit: naming snake case vs camel

File: Arduino/libraries/sensors/sensors.cpp [hunk 2]
```diff
@@ -285,4 +313,78 @@ int16_t * Sensors::readIMU(MPU6050 accelgyro){
...
+  case ICM20948_ACCEL_RANGE_8_G:
+    Serial.println("+-8G");
+    break;
+  case ICM20948_ACCEL_RANGE_16_G:
+    Serial.println("+-16G");
+    break;
+  }
+  Serial.println("OK");
+  // icm.setGyroRange(ICM20948_GYRO_RANGE_2000_DPS);
+  Serial.print("Gyro range set to: ");
```
Comment (ParthSareen) [inline Arduino/libraries/sensors/sensors.cpp:342]: do we need to keep this?

File: Arduino/libraries/sensors/sensors.cpp [hunk 3]
```diff
@@ -285,4 +313,78 @@ int16_t * Sensors::readIMU(MPU6050 accelgyro){
...
+  sensors_event_t temp; 
+  sensors_event_t mag; 
+  icm->getEvent(&accel, &gyro, &temp, &mag);
+  icmReadings[0] = accel.acceleration.x; 
+  icmReadings[1] = accel.acceleration.y; 
+  icmReadings[2] = accel.acceleration.z; 
+  icmReadings[3] = gyro.gyro.x;
+  icmReadings[4] = gyro.gyro.y;
+  icmReadings[5] = gyro.gyro.z;
+  return;
```
Comment (ParthSareen) [inline Arduino/libraries/sensors/sensors.cpp:379]: very clean

File: Arduino/libraries/sensors/sensors.h [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    int _num_sensors;
+    int* _echo_pin;
+    uint16_t measurement_delay_us;
+    long previousMillis; // will store last time LED was updated
+    long interval; // interval at which to blink (milliseconds), 1 ms shorter than desired (time to finish processing)
+    long dt; // change in time actual (milliseconds)
+    float minGyroValue; // min +/-Gyro value, (converted to rad/s)
+};
+
+#endif
```
Comment (ParthSareen) [inline Arduino/libraries/sensors/sensors.h]: nit: newline at eof

File: Arduino/libraries/sensors/sensors.h [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    int * calibrateIMU(MPU6050 accelgyro);
+    void calibrateICM(Adafruit_ICM20948* icm);
+    void readICM(Adafruit_ICM20948* icm,  float icmReadings[6]);
+
+  private:
+    int _trig_pin;
+    int _num_sensors;
+    int* _echo_pin;
+    uint16_t measurement_delay_us;
+    long previousMillis; // will store last time LED was updated
```
Comment (ParthSareen) [inline Arduino/libraries/sensors/sensors.h]: camel vs. snake same as prev

File: Arduino/libraries/sensors/sensors.h [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+#include "Adafruit_ICM20X.h"
+#include "Adafruit_ICM20948.h"
+#include "Adafruit_Sensor.h"
+class Sensors {
+  public:
+    Sensors(int trig_pin, int echo_pin[3], int num_sensors);
+    void readUltrasonic(HCSR04 *hc, float usReadings[3]);
+    int16_t * readIMU(MPU6050 accelgyro);
+    bool calibrateUltrasonic(int num_readings, HCSR04 hc);
+    int * calibrateIMU(MPU6050 accelgyro);
```
Comment (ParthSareen) [inline Arduino/libraries/sensors/sensors.h:16]: probably wanna delete these when we are sure we are not using it

**Review by ParthSareen — APPROVED** [review]

File: Arduino/libraries/MPU6050/MPU6050.h [hunk 7]
```diff
@@ -0,0 +1,1046 @@
+// I2Cdev library collection - MPU6050 I2C device class
```
Comment (andreachakma) [inline Arduino/libraries/MPU6050/MPU6050.h:1]: Probably could get rid of all these MPU6050 Files

File: Arduino/libraries/sensors/sensors.cpp (hunk 2 above)
Comment (andreachakma) [reply Arduino/libraries/sensors/sensors.cpp:342]: Agreed this could probably be removed.

**Review by andreachakma — APPROVED** [review]
looks good - apart from the MPU6050 header/functions that need to be deleted when we're sure we won't need them.

File: Arduino/libraries/sensors/sensors.cpp (hunk 3 above)
Comment (ShyamHari) [reply Arduino/libraries/sensors/sensors.cpp:379]: LOL whats simple is best

File: Arduino/libraries/sensors/sensors.h (hunk 6 above)
Comment (ShyamHari) [reply Arduino/libraries/sensors/sensors.h:16]: Gonna save this for a later time if we find our current IMU (ICM20948) isn't working. Will keep for now.

File: Arduino/libraries/MPU6050/MPU6050.h (hunk 7 above)
Comment (ShyamHari) [reply Arduino/libraries/MPU6050/MPU6050.h:1]: Saving until we're sure this new IMU works.

**Pull request merged**